Samples/certificados.php

```diff
@@ -0,0 +1,323 @@
+<?php 
+
+class Certificados{
+	private $_path = '';
+	public $_keyPem = '';
+	public $_cerPem = '';
+	public $_pfx = '';
+	private $_return = array();
+	
+	function __construct($pathCertificados = null){
+            $this->_path = $pathCertificados;
+	}
+	
+	private function _estableceError($result, $mensajeError = null, $arrayExtras = null){
+            $this->_return = array();
+            $this->_return['result'] = $result;
+            if($mensajeError != null){
+                    $this->_return['error'] = $mensajeError;
+            }
+            if($arrayExtras != null){
+                    foreach ($arrayExtras as $key => $val){
+                            $this->_return[$key] = $val;
+                    }
+            }
+	}
+	
+        function isEnabled($func) {
+            return is_callable($func) && false === stripos(ini_get('disable_functions'), $func);
+        }
+                
+	public function generaKeyPem($nombreKey, $password){
+	
+            $nombreKey = $this->_path.$nombreKey;
+
+            if (file_exists($nombreKey)) {
+                $salida = shell_exec('openssl pkcs8 -inform DER -in '.$nombreKey.' -out '.$nombreKey.'.pem -passin pass:'.$password.' 2>&1');
+                if($salida == '' || $salida == false || $salida == null){
+                    $this->_keyPem = $nombreKey.'.pem';
+                    $this->_estableceError(1);
+                    return $this->_return;
+                }else if (strpos($salida, 'Error decrypting') !== false) {
+                    $this->_estableceError(0, 'Contraseña incorrecta');
+                    return $this->_return;
+                }else {
+                    $this->_estableceError(0, 'No se logro generar el key.pem');
+                    return $this->_return;
+                }
+
+            }else {
+                $this->_estableceError(0, 'El archivo requerido no esta disponible');
+                return $this->_return;
+            }
+	}
+        
+	public function generaKeyPemPrivado($nombreKey){
+            
```

Comment:
Qué pasa si no se pasa no existe el archivo?

Reply:
nos enviará el error siguiente: $this->_estableceError(0, 'No se logro generar el key.pem'); con file_exists verificas la existencia

Reply:
Se elimina este fragmento de código debido a que debe ir en el toolkit